pkg/asset/installconfig/gcp/client.go

```diff
@@ -139,9 +139,9 @@ func (c *Client) GetDNSZoneByName(ctx context.Context, project, zoneName string)
 
 }
 
-// GetPublicDNSZone returns a public DNS zone for a basedomain.
-func (c *Client) GetPublicDNSZone(ctx context.Context, project, baseDomain string) (*dns.ManagedZone, error) {
-	ctx, cancel := context.WithTimeout(context.TODO(), 1*time.Minute)
+// GetDNSZone returns a DNS zone for a basedomain.
+func (c *Client) GetDNSZone(ctx context.Context, project, baseDomain string, isPublic bool) (*dns.ManagedZone, error) {
```

Comment:
NIT: a `bool` is fine but it would be nicer if we defined a proper enum type making it obvious in the code if the return matches the intention:
```suggestion
func (c *Client) GetDNSZone(ctx context.Context, project, baseDomain string, zoneType DNSZoneType) (*dns.ManagedZone, error) {
```
```golang
type DNSZoneType int
const (
    DNSZonePublic DNSZoneType = iota
    DNSZonePrivate
)

[...]
    publicZone, err := client.GetDNSZone(ctx, installConfig.Config.GCP.ProjectID, installConfig.Config.BaseDomain, DNSZonePublic)
[...]
    privateZone, err := client.GetDNSZone(ctx, installConfig.Config.GCP.ProjectID, installConfig.Config.ClusterDomain(), DNSZonePrivate)
```
Hopefully it'd make easier to catch a mistake like: 
```golang
    publicZone, err := client.GetDNSZone(ctx, installConfig.Config.GCP.ProjectID, installConfig.Config.BaseDomain, DNSZonePrivate)
```

Reply:
I will keep this one as a boolean for now. We can readdress in a new PR if this is a problem. This creates a cyclic import for the installer mock client. This could reside in the types but it wont be used anywhere else.